homeassistant/components/smarttub/config_flow.py

```diff
@@ -51,3 +51,16 @@ async def async_step_user(self, user_input=None):
             return self.async_abort(reason="reauth_successful")
 
         return self.async_create_entry(title=user_input[CONF_EMAIL], data=user_input)
+
+    async def async_step_reauth(self, user_input=None):
+        """Get new credentials if the current ones don't work anymore."""
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(self, user_input=None):
+        """Dialog that informs the user that reauth is required."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="reauth_confirm",
+                data_schema=vol.Schema({}),
```

Comment:
How would the end-user know which account it to use? As this integration supports multiple instances, there is no indication on which account one has to reauth with.

Right now, if I log in with the wrong account, the wrong accounts gets updated (because `existing_entry = await self.async_set_unique_id(account.id)` fetches whatever matches). Leaving me with the impression it was successful, while in fact, it wasn't.

Reply:
Thanks for the review. I think I should do two things:

1. Display the (existing/old) account email address in the form, to give the user a hint as to which account we're talking about
2. When the user provides updated credentials, verify that the account id matches, and if not, abort

Does that sound right?

Reply:
That sounds right 👍

Reply:
Where is a good place to store the account ID? Normally it's retrieved from the API when we authenticate, but if the authentication fails then we don't get an account ID. So we need to persist it somewhere in order to check whether the new credentials refer to the same account or not.

Reply:
The account ID is the unique ID for the config entry. Once you retrigger the auth in the controller, you could send it along to the config flow. That way, the flow has a reference to the entry that triggered it.

Reply:
Thanks, I used that approach